### PR TITLE
#1367: Scalar - Replacing usage of MatcherAssert.assertThat

### PR DESCRIPTION
**Pull request by marceloamadeu**
This PR is for #1367

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#1371](https://github.com/yegor256/cactoos/issues/1371) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/1158dafb-cefe-4645-beeb-19b0c497220f); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/scalar/BoolOfTest.java [hunk 1]
```diff
@@ -37,28 +37,28 @@
...
     @Test
     public void trueTest() throws Exception {
-        MatcherAssert.assertThat(
-            "Can't parse 'true' string",
+        new Assertion<>(
+            "Must be parsed string 'true'",
             new BoolOf("true").value(),
-            Matchers.equalTo(true)
-        );
+            new IsEqual<>(true)
```
Comment (victornoel) [inline src/test/java/org/cactoos/scalar/BoolOfTest.java:43]: @marceloamadeu can you replace all those `new BoofOf(xxx).value()` with `new BoofOf(xxx)` and use `ScalarHasValue` as the matcher?

File: src/test/java/org/cactoos/scalar/ConstantTest.java [hunk 2]
```diff
@@ -37,20 +38,20 @@
...
     public void returnsGivenValue() throws Exception {
         final String value = "Hello World";
-        MatcherAssert.assertThat(
-            "Can't return given value",
+        new Assertion<>(
+            "Must return given value",
             new Constant<>(value).value(),
-            Matchers.equalTo(value)
-        );
+            new IsEqual<>(value)
```
Comment (victornoel) [inline src/test/java/org/cactoos/scalar/ConstantTest.java:44]: @marceloamadeu same here with `ScalarHasValue`

File: src/test/java/org/cactoos/scalar/EqualsTest.java [hunk 3]
```diff
@@ -37,50 +37,50 @@
...
-            "Can't compare if two integers are equals",
+        new Assertion<>(
+            "Must compare if two integers are equals",
             new Equals<>(
                 () -> Integer.valueOf(1),
                 () -> Integer.valueOf(1)
             ).value(),
-            Matchers.equalTo(true)
-        );
+            new IsEqual<>(true)
```
Comment (victornoel) [inline src/test/java/org/cactoos/scalar/EqualsTest.java:46]: @marceloamadeu here too

File: src/test/java/org/cactoos/scalar/FoldedTest.java [hunk 4]
```diff
@@ -39,29 +39,29 @@
...
             new Folded<>(
                 0L, (first, second) -> first + second,
                 new HeadOf<>(
                     10,
                     new RangeOf<>(0L, Long.MAX_VALUE, value -> ++value)
                 )
             ).value(),
-            Matchers.equalTo(45L)
-        );
+            new IsEqual<>(45L)
```
Comment (victornoel) [inline src/test/java/org/cactoos/scalar/FoldedTest.java:51]: @marceloamadeu here too

**Comment by marceloamadeu** [comment]
@victornoel `IsEqual` was replaced with `ScalarHasValue` and `TODO` was added. Please check that it is correct.
Tks

**Comment by victornoel** [comment]
@marceloamadeu very nice, thank you!

@paulodamaso looks good to me

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**